conan/tools/cmake/cmakedeps/templates/macros.py

```diff
@@ -0,0 +1,88 @@
+import textwrap
+
+from conan.tools.cmake.cmakedeps.templates import CMakeDepsFileTemplate
+
+"""
+
+cmakedeps_macros.cmake
+
+"""
+
+
+class MacrosTemplate(CMakeDepsFileTemplate):
+    """cmakedeps_macros.cmake"""
+
+    def __init__(self):
+        super(MacrosTemplate, self).__init__(req=None, configuration=None)
+
+    @property
+    def filename(self):
+        return "cmakedeps_macros.cmake"
+
+    @property
+    def context(self):
+        return {}
+
+    @property
+    def template(self):
+        return textwrap.dedent("""
+        function(conan_message MESSAGE_OUTPUT)
+            if(NOT CONAN_CMAKE_SILENT_OUTPUT)
+                message(${ARGV${0}})
+            endif()
+        endfunction()
+
+       macro(conan_find_apple_frameworks FRAMEWORKS_FOUND FRAMEWORKS FRAMEWORKS_DIRS)
+           if(APPLE)
+               foreach(_FRAMEWORK ${FRAMEWORKS})
+                   message("BUSCANDO FRAMEWORK: ${_FRAMEWORK}")
+                   # https://cmake.org/pipermail/cmake-developers/2017-August/030199.html
+                   find_library(CONAN_FRAMEWORK_${_FRAMEWORK}_FOUND NAME ${_FRAMEWORK} PATHS ${FRAMEWORKS_DIRS} CMAKE_FIND_ROOT_PATH_BOTH)
+                   if(CONAN_FRAMEWORK_${_FRAMEWORK}_FOUND)
+                       list(APPEND ${FRAMEWORKS_FOUND} ${CONAN_FRAMEWORK_${_FRAMEWORK}_FOUND})
+                       message("Framework found! ${FRAMEWORKS_FOUND}")
+                   else()
+                       message(FATAL_ERROR "Framework library ${_FRAMEWORK} not found in paths: ${FRAMEWORKS_DIRS}")
```

Comment:
What if users use it for system frameworks? Can we have the same issues as libs-system_libs?

Reply:
We have tests using system frameworks like: `self.cpp_info.frameworks.append("Foundation")` and it is finding them.

Reply:
Note: There is no change in that code since the old generators, so it seem to be working